themes/default/content/blog/launch-redshift-into-aws-vpc-with-pulumi/index.md

```diff
@@ -0,0 +1,303 @@
+    assumeRolePolicy: JSON.stringify({
+        Version: "2012-10-17",
+        Statement: [{
+            Action: "sts:AssumeRole",
+            Effect: "Allow",
+            Sid: "",
+            Principal: {
+                Service: "redshift.amazonaws.com",
+            },
+        }],
+    }),
+    inlinePolicies: [{
+        name: "pulumi-policy",
+        policy: JSON.stringify({
+            Version: "2012-10-17",
+            Statement: [{
+                Action: ["s3:Get*", "s3:List*"],
+                Effect: "Allow",
+                Resource: "*",
+            }],
+        }),
+    }]
+});
+```
+
+Place those declarations after the statements related to the AWS S3 Bucket but before the statements declaring the Amazon Redshift cluster.
+
+We declare the AWS IAM role similar to the AWS console. We use a JSON-formatted object containing version and statement attributes. As with all other AWS services, the version is the publication date. The statement contains the action, effect, and principal. Here, we set those to `sts:AssumeRole`, `Allow`, and `redshift.amazonaws.com`, respectively. Then, we define the inline policy, which enables the principal to have read-only access to Amazon S3.
+
+Deploy those resources by typing:
+
+```bash
+$ pulumi up
+```
+
+We can then see the new role under the IAM console:
+
+![Screenshot of the AWS console showing the newly provisioned IAM role](./aws-console-6-iam-role.png)
+
+## Updating the Amazon Redshift Cluster with the AWS IAM Role
+
+In the last step, we assign the AWS IAM role to the Amazon Redshift Cluster. We add the `iamRoles` attribute to the Amazon Redshift cluster declaration:
+
+```typescript
+// Redshift cluster
+const redshiftCluster = new aws.redshift.Cluster("pulumi-redshift-cluster", {
+    clusterIdentifier: "pulumi-redshift-cluster",
+    clusterType: "single-node",
+    databaseName: "tempdb",
+    masterPassword: "Passw0rD",
+    masterUsername: "admin",
+    nodeType: "dc2.large",
+    clusterSubnetGroupName: subnetGroup.name,
+    skipFinalSnapshot: true,
+    iamRoles: [iamRole.arn] // <-- Add this property.
+});
+```
+
+Specify the roles using their Amazon Resource Names (ARNs). Here, we use just one role and obtain its ARN by reading the arn property of the object created previously.
+
+We can now update the cluster to confirm that we assigned the AWS IAM role to the cluster:
+
+```bash
+$ pulumi up
+```
+
+To see the IAM role, go to the Properties tab of the cluster and scroll down to Associated IAM roles:
+
+![Screenshot of the AWS console showing the IAM role associated with the Redshift cluster](./aws-console-7-cluster-iam-roles.png)
+
+After preparing the infrastructure, we can now use it to process data. The first step is to load data from the S3 source, [as demonstrated in the AWS documentation](https://docs.aws.amazon.com/redshift/latest/dg/tutorial-loading-data.html).
+
+Now remove all deployed cloud resources using the single command:
```

Comment:
The word "Now" is used 10 times in this article. Perhaps here it would be better to say something like "When you are finished or no longer need this Redshift cluster, you can remove..."

Reply:
Yep, agree here too. Feel free to submit any suggestions you like here and I'll apply 'em.